cheatsheets/gleam-for-scala-users.md

```diff
@@ -0,0 +1,1319 @@
+keys and values.
+
+In Gleam, maps can have keys and values of any type, but all keys must be of
+the same type in a given map and all values must be of the same type in a
+given map. The type of key and value can differ from each other.
+
+There is no map literal syntax in Gleam, and you cannot pattern match on a map.
+Maps are generally not used much in Gleam, custom types are more common.
+
+#### Scala
+
+```scala
+Map("key1" -> "value1", "key2" -> "value2")
+Map("key1" -> "1", "key2" -> 2) // compiles, but infers an unexpected type
+val myMap: Map[String, String] = Map("key1" -> "1", "key2" -> 2) // type error
+```
+
+
+#### Gleam
+
+```gleam
+import gleam/map
+
+map.from_list([#("key1", "value1"), #("key2", "value2")])
+map.from_list([#("key1", "value1"), #("key2", 2)]) // Type error!
+```
+
+### Numbers
+
+Scala has 8-bit, 32-bit, and 64-bit signed integers (`Byte`, `Short`, `Int`, `Long`)
+and also 32-bit and 64-bit floating point numbers (`Float`, `Double`). For arbitrary
+sized numbers, there are `BigDecimal` and `BigInteger`.
+
+Gleam has both support `Integer` and `Float`. Integer and Float sizes for
+both depend on the platform: JavaScript and Erlang.
+
+#### Scala
+
+Scala will automatically cast integers into floating point numbers when doing
+math operations between them.
+
+```scala
+1 / 2 // 0
+1.5 + 10 // 11.5
+```
+
+#### Gleam
+
+```gleam
+1 / 2 // 0
+1.5 + 10 // Compile time error
+```
+
+You can use the gleam standard library's `int` and `float` modules to convert
+between floats and integers in various ways including `rounding`, `floor`,
+`ceiling` and many more.
+
+## Flow control
+
+### Case
+
+Case is one of the most used control flow in Gleam. It can be seen as a switch
+statement on steroids. It provides a terse way to match a value type to an
+expression. It is also used to replace `if`/`else` statements.
+
+Scala and Gleam both support guards, destructuring and disjoint union matching.
+
+#### Scala
+
+Scala has both `if`/`else` and `match` expressions.
+
+```scala
+def httpErrorImpl1(status: Int) = {
+  if (status == 400) {
+      "Bad request"
+  } else if (status == 404) {
+      "Not found"
+  } else if (status == 418) {
+      "I'm a teapot"
+  } else {
+    "Internal Server Error"
+  }
+}
+
+def httpErrorImpl2(status: Int): String = {
+  status match { 
+    case 400 => "Bad request"
+    case 404 => "Not found"
+    case 418 => "I'm a teapot"
+    case _ => "Internal Server Error"
+  }
+}
+```
+
+Cases can also have guards, similar to Gleam
+```scala
+status match {
+  case 400 => "Bad request"
+  case 404 => "Not found"
+  case _ if status / 100 == 4 => "4xx"
+  case _ => "I'm not sure"
+}
+```
+
+The `|` operator works just like in Gleam.
+
+```scala
+number match {
+  case 2 | 4 | 6 | 8 => "This is an even number"
+  case 1 | 3 | 5 | 7 => "This is an odd number"
+  case _ => "I'm not sure"
+}
+```
+
+#### Gleam
+
+The case operator is a top level construct in Gleam:
```

Comment:
What's a top level construct?

Reply:
This is another section that was re-used from the [PHP guide](https://gleam.run/cheatsheets/gleam-for-php-users/#case), and it still has this language on the site today.

I agree it doesn't really add anything here, especially for a scala audience since case will already be very intuitive to them.